### PR TITLE
Issue 5847: Open visibility of AsyncBaseChunkStorage.OperationContext

### LINKED ISSUE #5847: AsyncBaseChunkStorage.OperationContext class has package level visibility instead of being protected.
**Describe the bug**
AsyncBaseChunkStorage.OperationContext class has package level visibility instead of being protected.
This prevents third party ChunkStorage implementations  from inheriting directly from AsyncBaseChunkStorage

**To Reproduce**
Inherit directly from AsyncBaseChunkStorage

**Expected behavior**
OperationContext is either a public class or is exposed as protected.
In other words no compile errors.
**Screenshots**
Not applicable

**Additional context**

As a side note : I wonder why this is not a compiler error or why spotbugs didn't complain. The protected members have instance of OperationContext as parameters.

### PR DESCRIPTION
**Pull request by diegosalvi**
**Change log description**  
Visibility change of AsyncBaseChunkStorage.OperationContext to protected

**Purpose of the change**  
Fixes #5847.

**What the code does**  
Just a visibility change of AsyncBaseChunkStorage.OperationContext from default to protected

**How to verify it**  
Implement AsyncBaseChunkStorage outside io.pravega.segmentstore.storage.chunklayer package

### CODE REVIEW

**Comment by eolivelli** [comment]
@diegosalvi  I believe you have to "sign" your commits in order to make DCO checker happy.

Also please create an issue, in Pravega we are using github issues in order to create the release notes

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/AsyncBaseChunkStorage.java [hunk 1]
```diff
@@ -712,7 +711,7 @@ protected void checkChunkName(String chunkName) {
     /**
      * Context data used to capture operation runtime information.
      */
-    static class OperationContext {
+    protected static class OperationContext {
```
Comment (eolivelli) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/AsyncBaseChunkStorage.java:714]: can we add a comment that explains that this class is intended to be used by subclasses and how ?

**Review by eolivelli — CHANGES_REQUESTED** [review]

**Review by andreipaduroiu — CHANGES_REQUESTED** [review]
@diegosalvi Thank you for your contribution!

I see this is your first PR on Pravega. As you probably already observed, we have some checks enabled on our repository that must pass before merging.

Please refer to our contribution guidelines (https://github.com/pravega/pravega/wiki/Contributing#issuesandprs), which explains how to create PRs (and associated issues) and how to sign your commits.

**Comment by fpj** [comment]
Is there an issue associated to this PR?

**Comment by sachin-j-joshi** [comment]
Created an Issue.  @diegosalvi  Please use this in commit description and PR comments.
https://github.com/pravega/pravega/issues/5847

**Comment by diegosalvi** [comment]
Changed commit descrition to match issue #5847 and signed off.

**Comment by diegosalvi** [comment]
Hi, I think I've added all requested changes, I'm missing anything?

**Review by sachin-j-joshi — APPROVED** [review]
LGTM

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by sachin-j-joshi** [comment]
@andreipaduroiu are we good to go ?

**Review by andreipaduroiu — APPROVED** [review]
Thank you for creating an issue. In the future, please follow our contribution guidelines and pull request guidelines - in particular, how to reference issues from PRs.

I have fixed this PR for you (Title + `Fixes`). Good to go.

**Pull request merged**